src/effection/events/mailbox.ts

```diff
@@ -0,0 +1,46 @@
+import { fork, monitor, Operation } from 'effection';
+import { EventEmitter } from 'events';
+
+import { on, onEach, EventName } from '../events';
+import { compile } from './pattern';
+export { any } from './pattern';
+
+export class Mailbox {
+  private subscriptions = new EventEmitter();
+  private messages = new Set();
+
+  *send(message: unknown): Operation {
+    this.messages.add(message);
+    this.subscriptions.emit('message');
+  }
+
+  *receive(pattern: unknown = undefined): Operation {
+    let match = compile(pattern);
+    let { messages, subscriptions } = this;
```

Comment:
Do we actually need this to close over these variables? It feels to me like `this.messages` is clearer than just `messages`, when seen in isolation.

Reply:
Not really, I'll explicitly refer to `this`